Packs/CommonScripts/Scripts/IsDomainInternal/IsDomainInternal.py

```diff
@@ -0,0 +1,93 @@
+from time import sleep
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+POLLING_TIME = 2  # Interval to wait in seconds when polling to check if indicator was created
+
+
+def check_domain_internal(domain, internal_domains):
+    parts = domain.split(".")
+    for i in range(len(parts), 0, -1):
+        sub = ".".join(parts[-i:])
+        if sub in internal_domains:
+            return True
+    return False
+
+
+def create_markdown_table(domain_list):
+    domain_list = sorted(domain_list, key=lambda x: not x['Internal'])
+    markdown_string = "| Domain Name | Internal |\n| --- | --- |\n"
+    for domain in domain_list:
+        markdown_string += f"| {domain['Name']} | {domain['Internal']} |\n"
+    return markdown_string
+
+
+internal_domains_list = demisto.args().get("InternalDomainsListName", "InternalDomains")
+domains_to_check = argToList(demisto.args().get("Domains", None))
+
+# Get the list of internal domains from the XSOAR list:
+internal_domains = demisto.executeCommand("getList", {"listName": internal_domains_list})[0]['Contents']
+if "Item not found" in internal_domains:
+    return_error(f"The list name {internal_domains_list} does not exist.")
+
+# Split internal domains from XSOAR list to be a list:
+if internal_domains:
+    try:
+        internal_domains = internal_domains.split("\n")
+    except Exception as ex:
+        return_error(f"Could not parse the internal domains list. Please make sure that the list contains domain names, separated by new lines.\nThe exact error is: {ex}")
+else:
+    demisto.results("No internal domains were specified.")
+    quit()
```

Comment:
what does this do?

Reply:
If the XSOAR list of internal domains exists but is empty, I want the script to quit. What can I replace quit with?

Reply:
We usually don't write scripts in a global scope, but use methods, and then a `return` is possible, see [here](https://xsoar.pan.dev/docs/tutorials/tut-integration-ui)

Reply:
Moved code into main() function, changed quit to return